cpu/sam0_common/Makefile.include

```diff
@@ -1,11 +1,20 @@
 # Define the CPU family so we can differentiate between them in the code
 CFLAGS += -DCPU_FAM_$(shell echo $(CPU_FAM) | tr 'a-z-' 'A-Z_')
 
+# Set ROM and RAM lengths according to CPU model
+ifneq (,$(filter samd21g18a samd21j18a saml21j18a samr21g18a,$(CPU_MODEL)))
```

Comment:
This is fine here for now, but you could also try to use a wildcard or parse the size from the model number if you're feeling experimental. I'm guessing the 18 in the model is the rom size? (2**18 = 0x40000 = 256 KB)

Reply:
You're right. All SAMXxxY18 model have 256KB of flash.

Reply:
I thought about this but my skills in `make` made me feel insecure. Besides the fact that I think is better to explicitly state which are the memory capabilities of a given CPU model.

If you really want this I'll ask my "make guru" colleagues how to do it.

Reply:
no, like I wrote above, this is fine here now. This can be refactored later.

Reply:
BTW SAML21J18B is missing from the line.

Reply:
Do we support such mcu?

Reply:
Yes, SAML21-XPRO board can be equipped with SAML21J18A or SAML21J18B. I also heard Rev C is on the way.
SAML21J18A were only early engineering sample, Atmel does NOT sell them. Only SAML21J18B are sold.

Reply:
Oh, I didn't find it in the current base code, however it is in the vendor headers. I'll add it.